draft.rb

```diff
@@ -0,0 +1,23 @@
+# rubocop:disable all
+# SCRATCHPAD TODO: Remove this file
+
+Datadog.configure do |c|
+  c.tracer sampler: Datadog::Sampling::RuleSampler.new(
+    [
+      Datadog::Sampling::SimpleRule.new(name: 'operation.name', sample_rate: 0.9),
+      Datadog::Sampling::SimpleRule.new(service: 'service-1', sample_rate: 0.9),
+      Datadog::Sampling::SimpleRule.new(sample_rate: 0.7) { |span| span.name != 'important.operation' },
+      Datadog::Sampling::SimpleRule.new(sample_rate: 1.0)
+    ],
+    Datadog::Sampling::TokenBucket.new(1),
+    Datadog::PrioritySampler.new(
+      post_sampler: Datadog::RateByServiceSampler.new(
+        1.0,
+        env: proc { Datadog.tracer.tags[:env] } # TODO: how do I provide `tracer.tags`? Seems like a circular reference here.
```

Comment:
Initializing a fallback `PrioritySampler` is not trivial.
Maybe we could use an interface in our tracer that allows us to append or prepend a sampler to a sampling chain, so that we could prepend our `RuleSampler` before the existing `PrioritySampler`, and avoid having all this complex initialization information outside of the tracer.